### ISSUE TITLE
allowsInlineMediaPlayback for inline video on iPhone

### ISSUE THREAD

**Issue by anjimi** [Simbul/baker#650]
This is related to https://github.com/Simbul/baker/issues/611 with the error on exiting fullscreen video.

Until this error is resolved we are trying to disable fullscreen video option in our app, before we submit for approval.

I have used a javascript video player and have successfully disabled fullscreen video option for iPad where videos play inline by default.

The iPhone defaults to always playing video fullscreen, after searching for a solution to this, there is a property which needs to be set for the webview - allowsInlineMediaPlayback.

Relevant on SO: http://stackoverflow.com/questions/5054560/can-i-avoid-the-native-fullscreen-video-player-with-html5-on-iphone-or-android

From http://developer.apple.com/library/ios/#documentation/uikit/reference/UIWebView_Class/Reference/Reference.html :

> allowsInlineMediaPlayback
> A Boolean value that determines whether HTML5 videos play inline or use the native full-screen controller.
> @property(nonatomic) BOOL allowsInlineMediaPlayback
> Discussion
> The default value on iPhone is NO.
> In order for video to play inline, not only does this property need to be set on the view, but the video element in the HTML document must also include the webkit-playsinline attribute.
> Availability
> Available in iOS 4.0 and later.
> Declared In
> UIWebView.h

I'm not sure where is the right place to set this property in Xcode, could anybody point me to where I would need to set this property for the webview in Baker?

**Comment by anjimi** [comment]
Ok, I found the place to add this parameter. In BakerViewController.m

I added the line:

> webView.allowsInlineMediaPlayback = YES;

to this function:

> – (void)setupWebView:(UIWebView *)webView {

and added webkit-playsinline="webkit-playsinline" to html5 video tags:

> <video controls="controls" webkit-playsinline="webkit-playsinline"

Videos now play inline.

Sadly although I'm using this player http://mediaelementjs.com/ which can also handle youtube videos, so you get the same player/controls whether video is included file or youtube, which works perfectly on iPad, the videos with included files play inline on iPhone, but the youtube ones are still playing in the native fullscreen player on iPhone.

So after all I think we will launch our app for iPad only until the issue #611 is resolved.
